### PR TITLE
Calendar meeting query (part 1)

### PR DESCRIPTION
**Pull request by isabelsang**
Implement query function by doing test driven development (for more information see [https://github.com/isabelsang/step/blob/master/walkthroughs/week-5-tdd/project/calendar-walkthrough.md](url) )
Passes 6/11 tests, other 5 in progress

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,92 @@
...
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    //Case in which the requested meeting has a duration that is too long/short
+    if(duration > TimeRange.WHOLE_DAY.duration() || duration < 0){
+        return Arrays.asList();
+    } 
+
+    //List of time ranges of other events that attendees are attending
+    List<TimeRange> timeRanges = new ArrayList<TimeRange>();
```
Comment (BearBurg) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:40]: Consider using set here because we want to avoid dupes.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,92 @@
...
+    TimeRange currentRange;
+    for(String person : attendees){
+        for(Event event: events){
+            currentRange = event.getWhen();
+            if(event.getAttendees().contains(person) && !timeRanges.contains(currentRange)){
```
Comment (BearBurg) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:45]: We don't need the second condition here if timeRanges is a set.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,92 @@
...
+                timeRanges.add(currentRange);
+            }
+        }
+    }
```
Comment (BearBurg) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:49]: Consider extracting this part as a helper method for better readability. Also practicing java stream here might be a good idea.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -15,9 +15,92 @@
...
+        //If earliest event starts at 00:00 then update startOfAvail 
+        //to be the end time of the first event
+        startOfAvail = nextTimeRange.end();
+
+        //First event is accounted for, so update lastTimeRange
+        lastTimeRange = nextTimeRange;
+        timeRangesCtr++;
+    } 
+
+    while(startOfAvail < TimeRange.END_OF_DAY && timeRangesCtr < timeRanges.size()){
```
Comment (BearBurg) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:76]: You could add a dummy first event (0,0) to handle some corner cases. Also you could add a dummy last event (24 * 60, 24 * 60) as the last event.

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java [hunk 5]
```diff
@@ -30,4 +30,14 @@ public void testGreeting() {
 
     Assert.assertEquals("Hello Ada", greeting);
```
[the remark this answers is not in the document]
Comment (isabelsang) [reply walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:31]: Just to double check, this means I should change "import org.junit.Assert;" to "import static org.junit.Assert;" and this will allow me to call assertEquals instead of Assert.assertEquals ?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (isabelsang) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:40]: I used a set at first but then switched it to an ArrayList so I could sort it in ascending order based on the start time of the TimeRange (see line 57).  This rest of the algorithm is based off of the list timeRanges being in the correct order, as it starts at the beginning of the day and iterates through this list to chronologically see when people are available. 
I just did some research though and found that there is a SortedSet.  I don't know much about this type of set... do you think it would be a good solution?

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (BearBurg) [reply walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:31]: Yes. 

```Static imports are appropriate when the usage is clear without additional context of the enclosing type. Static imports serve to remove redundant context and reduce boilerplate. The method/field should be fairly well-known, and self-describing.```

Here Assert.assertEquals are very well-known and self-describing so most of the time people will use static import.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (BearBurg) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:40]: You can use TreeSet as SortedSet. Or you could simply use hashSet to find all the timeRanges and add them all to a arrayList.

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (isabelsang) [reply walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:31]: Gotcha.  The makers of this walkthrough wrote  most of these test functions but good to know and I will make the changes!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 4 above)
Comment (isabelsang) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:76]: Would you be able to tell me more about this idea tomorrow during our sync please?
